src/Agent.Sdk/Util/LoggedSecretMasker.cs

```diff
@@ -0,0 +1,107 @@
+using Microsoft.TeamFoundation.DistributedTask.Logging;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Agent.Sdk.Util
+{
+    /// <summary>
+    /// Extended secret masker service, that allows to log origins of secrets
+    /// </summary>
+    public class LoggedSecretMasker : ILoggedSecretMasker
+    {
+        private ISecretMasker _secretMasker;
+        private ITraceWriter _trace;
+
+        private void Trace(string msg)
+        {
+            this._trace?.Info(msg);
+        }
+
+        public LoggedSecretMasker(ISecretMasker secretMasker)
+        {
+            this._secretMasker = secretMasker;
+        }
+
+        public void SetTrace(ITraceWriter trace)
+        {
+            this._trace = trace;
+        }
+
+        public void AddValue(string pattern)
+        {
+            this._secretMasker.AddValue(pattern);
+        }
+
+        /// <summary>
+        /// Overloading of AddValue method with additional logic for logging origin of provided secret
+        /// </summary>
+        /// <param name="value">Secret to be added</param>
+        /// <param name="origin">Origin of the secret</param>
+        public void AddValue(string value, string origin)
+        {
+            this.Trace($"Setting up value for origin: {origin}");
+            if (value == null)
+            {
+                this.Trace($"Value is empty.");
+                return;
+            }
+            this._secretMasker.AddValue(value);
```

Comment:
Nit: could it make sense to call AddValue(pattern) instead - to avoid duplication?

Reply:
Yep, I updated code by your suggestion